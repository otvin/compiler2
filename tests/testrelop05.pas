program testrelop05(output);
{relational operators with enumerated types}

type fruit = (apple, pear, banana, cherry, guava, mango);

var myfruit, myotherfruit:fruit;

begin
	myfruit := cherry;
	myotherfruit := pear;

	if myotherfruit < myfruit then
		writeln('pear is less than cherry');

	myfruit := pred(myfruit);

	if myfruit > myotherfruit then
		writeln('banana is greater than pear');

	myfruit := pred(myfruit);

	if myfruit > myotherfruit then
		writeln('oops!')
	else
		writeln('pear is not greater than pear');

	if myfruit = myotherfruit then 
		writeln('pear is equal to pear');

	myotherfruit := succ(succ(succ(myotherfruit)));

	if myotherfruit <> myfruit then
		writeln('guava and pear are not equal');

end.
