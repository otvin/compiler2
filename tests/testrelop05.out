pear is less than cherry
banana is greater than pear
pear is not greater than pear
pear is equal to pear
guava and pear are not equal
